src/main/kotlin/Board.kt

```diff
@@ -0,0 +1,108 @@
+class Board(initialPositions: Map<Char, List<String>> = mapOf()) {
```

Comment:
There was enough shared functionality and state that it made sense to me to break this into it's own class. That let's you sperate interface and main logic from logic and state that's specific to the board.